### PR TITLE
Pass LoginChainData and Skin to PlayerAsyncPreLoginEvent

### PR DESCRIPTION
**Pull request by Alemiz112**

### CODE REVIEW

File: src/main/java/cn/nukkit/Player.java [hunk 1]
```diff
@@ -2513,7 +2512,7 @@ public void onCompletion(Server server) {
                     break;
                 case ProtocolInfo.PLAYER_SKIN_PACKET:
                     PlayerSkinPacket skinPacket = (PlayerSkinPacket) packet;
-                    Skin skin = skinPacket.skin;
+                    skin = skinPacket.skin;
```
Comment (PetteriM1) [inline src/main/java/cn/nukkit/Player.java:2515]: This sets the skin even if the `isValid` below doesn't pass?

Comment (Alemiz112) [reply src/main/java/cn/nukkit/Player.java:2515]: No, it should reference to [this instance](https://github.com/PetteriM1/NukkitPetteriM1Edition/blob/9a6996c2e3d44f306d57875f3abae414fbbcba76/src/main/java/cn/nukkit/Player.java#L2419).

File: src/main/java/cn/nukkit/event/player/PlayerAsyncPreLoginEvent.java [hunk 2]
```diff
@@ -23,7 +25,8 @@ public static HandlerList getHandlers() {
 
     private final String name;
     private final UUID uuid;
-    private final String xuid;
+    private final LoginChainData chainData;
```
Comment (PetteriM1) [inline src/main/java/cn/nukkit/event/player/PlayerAsyncPreLoginEvent.java:28]: Shouldn't we have a getter for this too?

Comment (Alemiz112) [reply src/main/java/cn/nukkit/event/player/PlayerAsyncPreLoginEvent.java:28]: Solved

**Pull request merged**